query-overview.md

```diff
@@ -0,0 +1,122 @@
+---
+title: Overview
+description: Fetch feature values via queries.
+---
+
+---
+
+To request or compute data with Chalk, you'll use **queries**. In general,
+when you run a Chalk query, you are either: requesting the most up-to-date
+values for your feature sets, requesting a set of historical data
+for your feature sets, or running a backfill or batch job.
+
+The first use case is accomplished through **online queries**, which try to
+return values for a single feature set as quickly as possible: taking
+advantage of caching and distributed execution.
+
+The latter use cases are accomplished through offline queries which execute
+over the offline store and can return multiple instances of a feature set.
+
+## What are Queries
+
+At a high level, a query specifies input features and output features.
+Inputs differ slightly for online queries and offline queries, but in both
+cases the input must contain the primary key of your requested feature set.
+
+For example, for an online query, the input could be the id of a user for
+which you want to compute output features. You can also specify additional
+features that "overwrite" the features your resolver would otherwise compute.
+For instance, you could run an online query passing one of your user's ids but also
+a hard coded name. Using Chalk's CLI, this would look like the following:
+
+```sh
+chalk query --in user.id=1 --in user.name=mary --out user
+```
+
+For offline queries, the input is a list of ids for one of your feature
+sets and, optionally, any number of lists containing values for any other
+feature from the same namespace.
+
+Online and offline queries differ most in their output. An online
+query only ever returns one value for each requested output feature, which
+it retrieves either from the cache or by executing resolvers.
+
+An offline query returns all requested computed features for all requested ids.
+For instance, say you've computed the number of transactions (`num_transactions`)
+for a User with id 1 every day over the past week. Making an offline query
+with `user.id=1` as an input and `user.num_transactions` as an output would
+return all seven of the previously computed feature `num_transaction` feature
```

Comment:
"return all seven of the previously computed feature `num_transaction` feature values" sounds awkward because "feature" is repeated, might want to drop one of them or do something like "return all seven of the feature `num_transaction`'s values